### ISSUE TITLE
Ability to get the first emitted event only

### ISSUE THREAD

**Issue by gilm123** [CacheControl/json-rules-engine#221]
Hi, we are testing the use of the package for calculating a served version number, for each customer.
for example, rules are
- customers from `country-1` and the time is `5 AM` get `version 2.3`
- customer from `country-1` get `version 2.1`
- customer from `country-2` get `version 2.2`

user comes into the engine, -> `facts: { country: userCountry }`-> thrown into the engine.

on paper, it is looking promising, but we'd like to get 1 emitted result event( for `country-1` users for example). i could, in theory, take the first event out of the returned array, but when i have async dynamic facts, i cannot count on the order. 

saying "time" is a dynamic async fact i get from country fact, this calculation will result by the engine emitting the events that were calculated with "sync" facts first,
unless I'm giving priorities to the rules.

TL;DR
1. can i ask the engine to emit 1 event? the first successful event.
2. when having a combination of sync and async dynamic facts, can i know for sure I'll get the async "conditions" (facts) being calculated before the sync ones if i give them priorities? 

thank you

**Comment by CacheControl** [comment]
Hi, for order-specific logic, you can always rely on the engine to execute rule serially based their priority - so that might be a nature fit here. And yes, regardless of whether a rule has async logic, it will be executed to completion before a rule of lower priority is run.

If you want to avoid serial execution, another option would be to store the first event emission in a variable whose scope is outside of the rules engine, e.g.

```js
let firstEvent
engine.on('success', (event, almanac, ruleResult) => {
   if (!firstEvent) firstEvent = event
})
```
